src/object.c

```diff
@@ -989,6 +989,12 @@ int equalStringObjects(robj *a, robj *b) {
          * long is the same. */
         return a->ptr == b->ptr;
     } else {
+        if (a->encoding != OBJ_ENCODING_INT &&
```

Comment:
maybe using sdsEncodedObject to check is better?

Reply:
sdsEncodedObject() has two condition judgments, using OBJ_ENCODING_INT would be faster.

Reply:
now we only have three encoding types for object, `OBJ_ENCODING_RAW`, `OBJ_ENCODING_EMBSTR`, `OBJ_ENCODING_INT`, so we can `sdslen(o->ptr)` if not `OBJ_ENCODING_INT`, but if we add other encoding, it will not work well. Besides, we usually use `sdsEncodedObject` to check if we can call `sdslen(o->ptr)`, it's more straightforward. And i don't think using OBJ_ENCODING_INT to check has significant improvement than sdsEncodedObject.

what's more, why not optimize `compareStringObjectsWithFlags` if we want to avoid `memcmp`, that is more common

Reply:
the reason is here: https://github.com/redis/redis/pull/14062#discussion_r2099111596

Reply:
OK. but as i said above, i don't think it is a good idea to only check `!= OBJ_ENCODING_INT`

Reply:
I have no objection to this, but it would be best to do a benchmark again. @vitahlin please take a look.

Reply:
Sure, I’d be happy to. I’ll run a new benchmark shortly.